### ISSUE TITLE
Energy Dashboard adds solar production to Individual Device Usage

### ISSUE THREAD

**Issue by iWasAWizard** [home-assistant/frontend#21902]
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.
- [X] I have tried reproducing the issue in [safe mode](https://www.home-assistant.io/blog/2023/11/01/release-202311/#restarting-into-safe-mode) to rule out problems with unsupported custom resources.

### Describe the issue you are experiencing

A duplicate or extremely similar number of kWh of usage is being applied to the Individual Device Usage list from Solar Production in the Energy dashboard, resulting in the display of energy deficits that don't actually exist.

### Describe the behavior you expected

The Individual Device Usage list should not factor in solar power production as a part of total power consumption. The total power consumption, and thus the values displayed on the IDU graph should be using the "Energy Usage" value to define the total usage for a given time period.

### Steps to reproduce the issue

**Reproduce**
1. Add an entity to "Energy Usage" to display total energy used. I am using the "sensor.feed_through_consumed_energy" entity provided by the SPAN integration.
2. Add an entity to "Solar Production" to display total solar power produced. I am using the "sensor.envoy_[XYZ]_energy_production_today" entity provided by the Enphase Envoy integration.
3. Add multiple entities to "Individual Devices" to represent portions of total energy used. I am using circuit power consumption entities from each of the breakers on my SPAN panel.

**Verify**
1. Wait for values to equalize. 2 hours, per the Energy Dash Configuration directions.
2. Note the value of the "Solar Production" key for a given time block. For this example I will use 9.47kWh, as displayed above.
3. Note the value of the "Energy Usage" key for a given time block. For this example I will use 2.46kWh, as displayed above.
4. Note the total value of the "Individual Devices Detail Usage". It displays a bar graph showing energy consumption over time, as well as the deficit between tracked energy and total energy usage.

The number provided by step 4 will be incorrect, it displays the usage values for each individual item, but then stacks a number which is either identical, or extremely close to the "Solar Production" value for that time block on top as "Untracked consumption". This results in the appearance of a large data deficit, despite none actually being present.

### What version of Home Assistant Core has the issue?

2024.9.0

### In which browser are you experiencing the issue with?

Mozilla Firefox 129.0.2 Build 20240819150008

### Which operating system are you using to run this browser?

macOS Sonoma 14.6.1

### State of relevant entities

```txt
See attached screenshots.
```

### Additional information

![Screenshot 2024-09-05 at 12-50-06 Energy – Home Assistant](https://github.com/user-attachments/assets/c7d0fde2-dd30-4f88-9ed8-362798353ce5)
![Screenshot 2024-09-05 at 12-54-06 Energy – Home Assistant](https://github.com/user-attachments/assets/cf6b0ee2-dfc2-46ed-8048-5a1123d121ae)
![Screenshot 2024-09-05 at 12-54-31 Energy – Home Assistant](https://github.com/user-attachments/assets/44195b92-c318-4469-b822-38af6fe54057)
![Screenshot 2024-09-05 at 12-55-00 Energy – Home Assistant](https://github.com/user-attachments/assets/086984b6-f816-4869-91dc-177d16187a0e)

**Comment by karwosts** [comment]
You are producing solar, but it does not appear to go anywhere. Are you selling it back to the grid? Do you have a grid return sensor?

If you're buying 2 kWh from the power company, and producing 9 kWh, and you're not selling the power, then you must be consuming 11kWh. That's what that graph is trying to tell you.

**Comment by Baxxy13** [comment]
1:00 PM - 2:00PM
A: import from grid ~2,5kWh
B: produced by solar ~9.5kWh 
C: consumed by individiual devices ~2.5kWh

A + B - C = ~9.5kWh ...your untracked consumption

**Comment by iWasAWizard** [comment]
I see. Explained that way, it makes sense even if it doesn't match intuition. Thanks for that. My electrical company integration doesn't provide a metric for solar grid return, so I'll work on creating a value template for that or something. Thank you!

**Comment by karwosts** [comment]
If you're just selling all your solar and you don't consume any of it, then you can just paste your solar sensor directly into the grid return option and that should balance everything out.

**Comment by iWasAWizard** [comment]
For anyone that comes through here later, I wanted to include what I did to fix the problem.

As my friends above showed me, energy production that isn't accounted for is treated by the system as an indicator that you're consuming more energy than you actually are. Not that it is being returned to grid, like I originally thought. I found a simple value template written by a community member that will take the instantaneous difference between solar production and grid consumption, which spits out a return-to-grid total.

https://community.home-assistant.io/t/calculate-return-to-grid-from-consumption-and-solar-production/378464/12

You can add this by going to Settings > Devices and Services > Helper. Define a new Helper. Set "unit of measurement" to kWh, "device class" to Energy, and "state class" to Total. Then set that new entity as your return-to-grid source.

You still have to wait a couple of hours for this to equalize, because the numbers will be super weird. I'll update this if that does not improve.

**Comment by iWasAWizard** [comment]
I agree, the default behavior is not intuitive. And for your case it’s actively making your graph harder to interpret, with no real way around it. I think without modification to the logic, which is valid, the Untracked Consumption feature should be opt-in.

I’m going to re-open this issue for discussion.

**Comment by gn3sda** [comment]
I agree, more configuration options needed as the default is not intuitive outside of one type of install. The system only works with solar connected to the load side of your system (typically main panelboard). I just installed a line tap system (installed between main panelboard and meter) which is the common way to install in the US when the system is over a certain size. I found this thread trying to understand why HA assumed my consumption drastically increased. Unaccounted for energy is sent to grid, not consumed.

**Comment by CrallH** [comment]
This is a strange way of handle this data. 
I just found this issue in my setup (just have my solar running for two months). So all solar production that i use my self is addad as untracked consumption. But i track allmost all my consumtion ( i have ~0,5kwh untracked consumption in general every hour) now when my solar produces some electricity during the days (not enougt to export thou) it gets added as untracked consumption, but as stated allreday, i track allmost all consuption with individal meaters so the individal graph gets way of saying that i consume more energy than i really do. 

![Image](https://github.com/user-attachments/assets/b7ab1128-407a-469b-8b64-6740e49272b1)

**Comment by karwosts** [comment]
@CrallH - Where does the extra solar energy go if you're not exporting it?

**Comment by CrallH** [comment]
> [@CrallH](https://github.com/CrallH) - Where does the extra solar energy go if you're not exporting it?
> 
> HA has to assume it's being used because it isn't accounted for anywhere else.

You are correct! I did a brain fart.  I can't get my head around the consumption during this day, something in my house have consumed way more than it should!!

**Comment by Dennizz** [comment]
I was seeing similar weird issues. After digging deeper, I found that the P1 entities that I was using for reporting "Return to grid" were set up incorrectly. I was using Interval in stead of cumulative. Once I rectified that, everything started to make sense.

**Comment by Loic691** [comment]
Hi
Any news about this issue ?
